programs/Makefile

```diff
@@ -241,6 +247,10 @@ test/benchmark$(EXEXT): test/benchmark.c $(DEP)
 	echo "  CC    test/benchmark.c"
 	$(CC) $(LOCAL_CFLAGS) $(CFLAGS) test/benchmark.c   $(LOCAL_LDFLAGS) $(LDFLAGS) -o $@
 
+test/header_test$(EXEXT): test/header_test.cpp $(DEP)
```

Comment:
To make this test self sufficient perhaps execute config.pl with full config.

Reply:
Good idea, but that's up to test scripts such as `all.sh`, not the job of the build scripts.

Reply:
What do you mean by self sufficient?

Reply:
My idea was to have a make target that indicates whether the headers are compatible with C++. But general consensus here is do the config outside Makefile. Hence ignore this.